go/src/koding/klient/machine/index/cache.go

```diff
@@ -10,12 +10,27 @@ import (
 	"path/filepath"
 	"strings"
 	"time"
+
+	"koding/klient/machine/index/filter"
 )
 
 // TempIndexDirPrefix defines the prefix of temporary index file created by
 // Cached structure.
 const TempIndexDirPrefix = "koding_index_"
 
+// DefaultFilter defines a default filter used to process index caching.
+var DefaultFilter filter.Filter = filter.MultiFilter{
+	filter.OsSkip(filter.DirectorySkip(".Trash"), "darwin"),                                                                   // OSX trash directory.
+	filter.OsSkip(filter.DirectorySkip(".Trashes"), "darwin"),                                                                 // OSX trash directory.
+	filter.NewWithError(filter.PathSuffixSkip(".git/index.lock"), "git repository on remote is in use"),                       // git index lock file.
```

Comment:
i am not sure if we should have these kind of filters - we can break a git index and cause devs pain
ie: 
* rebase on remote 
* rebase fails
* index.lock stays
* we should not be able to continue

@szkl might comment on these internals better